app/redux/wallet/actions.js

```diff
@@ -285,23 +270,37 @@ export const sendTransaction = ({ dstAddress, amount, fee, note }: { dstAddress:
   }
 };
 
-export const addTransaction = ({ tx, accountPK }: { tx: Tx, accountPK?: string }): Action => (dispatch: Dispatch, getState: GetState): Dispatch => {
+export const addTransaction = ({ tx, accountPK }: { tx: Tx, accountPK?: string }): Action => async (dispatch: Dispatch, getState: GetState): Dispatch => {
   try {
     const { accounts, transactions, currentAccountIndex, walletFiles } = getState().wallet;
     const index = accountPK ? accounts.findIndex((account) => account.pk === accountPK) : currentAccountIndex;
     const updatedTransactions = { ...transactions, [index]: [tx, ...transactions[index]] };
-    fileSystemService.updateFile({ fileName: walletFiles[0], fieldName: 'transactions', data: updatedTransactions });
+    await fileSystemService.updateFile({ fileName: walletFiles[0], fieldName: 'transactions', data: updatedTransactions });
+    dispatch(setTransactions({ transactions: updatedTransactions }));
+  } catch (error) {
+    throw new Error(error);
+  }
+};
+
+export const updateTransaction = ({ tx, accountPK }: { tx: Tx, accountPK?: string }): Action => async (dispatch: Dispatch, getState: GetState): Dispatch => {
+  try {
+    const { accounts, transactions, currentAccountIndex, walletFiles } = getState().wallet;
+    const index = accountPK ? accounts.findIndex((account) => account.pk === accountPK) : currentAccountIndex;
+    const txIndex = transactions[index].findIndex((transaction: Tx) => tx.address === transaction.address);
+    const transactionsArray = transactions[index].map((transaction: Tx, idx: number) => (idx === txIndex ? { ...tx } : { ...transaction }));
```

Comment:
simplify and make more robust these two lines
it can be something like this
const updatedTransactions = { ...transactions, [index]: [...transactions[index].slice(0, txIndex), tx, ...transactions[index].slice(txIndex + 1)] };